src/lib/device-online-state.ts

```diff
@@ -0,0 +1,317 @@
+				if ('id' in msg) {
+					const { id, message } = msg;
+
+					return Bluebird.try(() => {
+						const { uuid, nextState } = JSON.parse(message) as {
+							uuid: string;
+							nextState: DeviceOnlineStates;
+						};
+
+						// raise and event for the state change...
+						switch (nextState) {
+							case DeviceOnlineStates.Timeout:
+								this.scheduleChangeOfStateForDevice(
+									uuid,
+									DeviceOnlineStates.Timeout,
+									DeviceOnlineStates.Offline,
+									API_HEARTBEAT_STATE_TIMEOUT_SECONDS, // put the device into a timeout state if it misses it's scheduled heartbeat window... then mark as offline
+								);
+								this.deviceTimeout(uuid);
+								break;
+							case DeviceOnlineStates.Offline:
+								this.deviceOffline(uuid);
+								break;
+							default:
+								captureException(
+									new Error(
+										`An unexpected value was encountered for the target device state: ${nextState}`,
+									),
+								);
+						}
+					}).finally(() =>
+						this.rsmq.deleteMessageAsync({
+							qname: DeviceOnlineStateManager.EXPIRED_QUEUE,
+							id,
+						}),
+					);
+				} else {
+					// no message! this is thrown to restart the consume process...
+					return Bluebird.delay(1000).thenThrow(
+						new NoMessagesToConsumeSignal(),
+					);
+				}
+			})
+			.catch(err => {
+				if (err instanceof NoMessagesToConsumeSignal) {
+					return;
+				}
+
+				captureException(
+					err,
+					'DeviceStateManager ==> Error when consuming queue',
+				);
+			})
+			.then(() => this.consume());
+	}
+
+	private scheduleChangeOfStateForDevice(
+		uuid: string,
+		currentState: DeviceOnlineStates,
+		nextState: DeviceOnlineStates,
+		delay: number, // in seconds
+	) {
+		// remove the old queued state...
+		return this.redis
+			.getAsync(`${DeviceOnlineStateManager.REDIS_NAMESPACE}:${uuid}`)
+			.then(value => {
+				const { id } = JSON.parse(value) as { id: string };
+
+				if (id) {
+					return this.rsmq.deleteMessageAsync({
+						qname: DeviceOnlineStateManager.EXPIRED_QUEUE,
+						id,
+					});
+				}
+
+				return 1;
+			})
+			.catch(noop) // ignore errors when deleting the old queued state and queue the new one...
+			.then(() =>
+				this.rsmq.sendMessageAsync({
+					qname: DeviceOnlineStateManager.EXPIRED_QUEUE,
+					message: JSON.stringify({
+						uuid,
+						nextState,
+					}),
+					delay,
+				}),
+			)
+			.then(id =>
+				this.redis
+					.setAsync(
+						`${DeviceOnlineStateManager.REDIS_NAMESPACE}:${uuid}`,
+						JSON.stringify({
+							id,
+							currentState,
+						}),
+					)
+					.then(() =>
+						this.redis.expireAsync(
+							`${DeviceOnlineStateManager.REDIS_NAMESPACE}:${uuid}`,
+							delay + 5, // add a small 5 second buffer to allow some wiggle room on top of the timeout window...
+						),
+					),
+			);
+	}
+
+	public captureEventFor(uuid: string, timeoutSeconds: number) {
+		// see if we already have a queued state for this device...
+		return (
```

Comment:
What is the paren for?

Reply:
Prettier. :man_shrugging:

Reply:
Could you give a try manually removing the brackets here again so it's `return this.redis...` and then reformat with an up to date prettier, I'm hoping prettier has improved this case in the 5 months since

Reply:
I tried, and checked we're using the latest version of `prettier` - it really does want it to look like this 👍